src/client/Microsoft.Identity.Client/ApiConfig/AbstractConfidentialClientAcquireTokenParameterBuilder.cs

```diff
@@ -12,19 +13,21 @@
 
 namespace Microsoft.Identity.Client
 {
-#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME && !MAC_BUILDTIME // Hide confidential client on mobile platforms
-
     /// <summary>
     /// Base class for confidential client application token request builders
     /// </summary>
     /// <typeparam name="T"></typeparam>
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
+    [System.ComponentModel.EditorBrowsable(EditorBrowsableState.Never)]  // hide confidentail client on mobile
```

Comment:
This attribute ensures that Visual Studio (and potentially other IDEs) do not display this type in the list of suggestions. 

When applied to a method, it hides the method.
When applied to a class, it hides all instance methods on that class (not the static ones)